### PR TITLE
[Onprem] Modify Sky Admin's Setup on Docs

### LINKED ISSUE #1082: [Onprem] Installation docs: git clone fails for ssh url
On the installation page, `git clone ssh://git@github.com/skypilot-org/skypilot.git` fails. We should use https:// after we have open sourced

image

image

### PR DESCRIPTION
**Pull request by michaelzhiluo**
Fixes #1082

### CODE REVIEW

File: docs/source/reference/local/setup.rst [hunk 1]
```diff
@@ -23,7 +23,7 @@ To install Ray and SkyPilot for all users, run the following commands on all loc
    $ sudo -H pip3 install ray[default]==1.13.0
 
    $ # SkyPilot requires python >= 3.6.
-   $ git clone ssh://git@github.com/skypilot-org/skypilot.git
+   $ git clone https://github.com/skypilot-org/skypilot.git
    $ cd skypilot
    $ sudo -H pip3 install -e .
```
Comment (concretevitamin) [inline docs/source/reference/local/setup.rst:28]: How about `sudo -H pip3 install skypilot` for onprem?  I imagine onprem is tougher to set up -- fewer commands lower setup overhead?

Comment (romilbhardwaj) [reply docs/source/reference/local/setup.rst:28]: +1, if pip install skypilot works then we should use that

Comment (michaelzhiluo) [reply docs/source/reference/local/setup.rst:28]: Sounds good, admin should only work with stable releases^^

**Review by concretevitamin — APPROVED** [review]
Nice @michaelzhiluo. Btw, does this fix the issue reported by @WoosukKwon, or will it be a separate change?

**Comment by michaelzhiluo** [comment]
@concretevitamin It will be a separate change... I need to do a deeper investigation to see why @WoosukKwon is encountering such an issue.

**Pull request merged**